### PR TITLE
Add ENV['PUMA_NO_RUBOCOP'] to disable RuboCop install, add mswin ci

### PR DESCRIPTION
**Pull request by MSP-Greg**
### Description

1. Setting `ENV['PUMA_NO_RUBOCOP']` disables installation and use of RuboCop.  Since we're now doing RuboCop as a separate workflow, the other workflows do not need to install RuboCop and it's dependencies.  This lightens the resolution Bundler needs to do, and makes the bundle caches smaller.  Note that the extension gem [jaro_winkler](https://rubygems.org/gems/jaro_winkler) is a dependency of RuboCop, and it will not compile on mswin.

2. Small changes to allow Puma to build using Ruby Windows mswin build. Add to Actions CI.

### Your checklist for this pull request
- [x] I have reviewed the [guidelines for contributing](../blob/master/CONTRIBUTING.md) to this repository.
- [ ] I have added (or updated) appropriate tests if this PR fixes a bug or adds a feature.
- [x] My pull request is 100 lines added/removed or less so that it can be easily reviewed.
- [ ] If this PR doesn't need tests (docs change), I added `[ci skip]` to the title of the PR.
- [ ] If this closes any issues, I have added "Closes `#issue`" to the PR description or my commit messages.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed, including Rubocop.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -11,7 +13,7 @@ gemspec = Gem::Specification.load("puma.gemspec")
 Gem::PackageTask.new(gemspec).define
 
 # Add rubocop task
-RuboCop::RakeTask.new
+RuboCop::RakeTask.new unless NO_RUBOCOP
```
Comment (dentarg) [inline Rakefile:16]: WDYT of this instead?

```ruby
begin
  require "rubocop/rake_task"
  RuboCop::RakeTask.new
rescue LoadError
end

**Review by dentarg — APPROVED** [review]

File: Rakefile (hunk 1 above)
Comment (dentarg) [reply Rakefile:16]: (for all of the changes to `Rakefile`, not only the above line)

File: ext/puma_http11/extconf.rb [hunk 2]
```diff
@@ -14,6 +14,8 @@
   found_ssl = if (!$mingw || RUBY_VERSION >= '2.4') && (t = pkg_config 'openssl')
     puts 'using OpenSSL pkgconfig (openssl.pc)'
     true
+  elsif have_library('libcrypto', 'BIO_read') && have_library('libssl', 'SSL_CTX_new')
+    true
```
Comment (dentarg) [inline ext/puma_http11/extconf.rb:18]: Is this change related to this PR?

Comment (MSP-Greg) [reply ext/puma_http11/extconf.rb:18]: This started as a PR about the mswin build, and it might be a bug in mkmf.  The RuboCop ENV variable involved the same code, so I added it.  An issue or PR mentioned the idea of having a way to skip RuboCop, but I couldn't find it.

But, almost all current OpenSSL implementations name the files 'libcrypto' and 'libssl'.  Older installs did not.  Older Windows OpenSSL used name 'libeay32' and 'ssleay32'.  Anyway, it is needed for mswin, and most other builds use pkgconfig.

**Review by dentarg — APPROVED** [review]
